client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HttpClient.ts

```diff
@@ -34,19 +34,11 @@ export class HttpClient implements IHttpClient {
                     resolve(xhr.response);
                 }
                 else {
-                    reject({
-                        status: xhr.status,
-                        statusText: xhr.statusText
-                    });
+                    reject(new Error(xhr.statusText));
                 }
             };
 
-            xhr.onerror = () => {
-                reject({
-                    status: xhr.status,
-                    statusText: xhr.statusText
-                });
-            };
+            xhr.onerror = () => reject(new Error(xhr.statusText));
```

Comment:
We could add tests for this in the end-to-end tests - there is a web server there and it is easy to request a non-existing page...